### PR TITLE
Switch auto UI scale to use screen DPI

### PR DESCRIPTION
**Pull request by Kiisu-Master**
- This makes auto scaling use screen DPI.
- Limits scale to a calculated max scale to avoid glitches when min size would be bigger than usable screen size.

This has to be tested on more devices. @aaronfranke
I only tested this on my phone and laptop, which both have low resolution screens. Different DPI though so it's tested a little bit and seems to work.

Previously it was making the default experience on phone even worse, because it made UI even smaller than it already was without automatic scale. I know phones arent really supported anyways, but I hope this fix also makes it better on desktop.

Edit: I did some testing on my laptop (1366x768,  DPI 96). Previous implementation would make UI too small on it. New one seems to work well. Also tested with the AMD virtual resulution to set it to higher resolutions (FHD, DPI 120), and it still works well.

### CODE REVIEW

File: src/GlobalSettings.gd [hunk 1]
```diff
@@ -249,24 +247,59 @@ func get_validity_color(error_condition: bool, warning_condition := false) -> Co
...
+			return 0.75
+		elif dpi <= 96:
+			return 1.0
+		elif dpi <=160:
+			return 1.25
+		elif dpi <= 240:
+			return 1.5
+		elif dpi <= 480:
+			return 1.75
+		elif dpi > 480:
```
Comment (aaronfranke) [inline src/GlobalSettings.gd:299]: Make it clear to the reader that at least one of these blocks will be executed when `dpi != 72`

```suggestion
		else:  # dpi > 480:
```

File: src/GlobalSettings.gd [hunk 2]
```diff
@@ -249,24 +247,59 @@ func get_validity_color(error_condition: bool, warning_condition := false) -> Co
...
+		if dpi < 72:
+			return 0.75
+		elif dpi <= 96:
+			return 1.0
+		elif dpi <=160:
+			return 1.25
+		elif dpi <= 240:
+			return 1.5
+		elif dpi <= 480:
+			return 1.75
```
Comment (aaronfranke) [inline src/GlobalSettings.gd:298]: These values are just a bit too low. `DisplayServer.screen_get_dpi(screen)` returns `163` on my system with 200% scaling, so I would expect `2.0` to be returned, but this code is returning `1.5`.

File: src/GlobalSettings.gd [hunk 3]
```diff
@@ -249,24 +247,59 @@ func get_validity_color(error_condition: bool, warning_condition := false) -> Co
...
+	
+	var usable_screen_size := Vector2(
+		DisplayServer.screen_get_usable_rect(DisplayServer.window_get_current_screen()).size
+		# Subtract window decoration size.
+		 - (window.get_size_with_decorations() - window.size)
+		)
+	
+	# How much can window content size be multiplied by before it extends over the usable screen size.
+	var diff :=  usable_screen_size / window.get_contents_minimum_size()
+	var max_scale := floorf(minf(diff.x, diff.y)*4.0)/4.0
```
[the remark this answers is not in the document]
Comment (Kiisu-Master) [reply src/GlobalSettings.gd:259]: It don't look like that floors the value (we dont want to round it up)

File: src/GlobalSettings.gd (hunk 2 above)
Comment (Kiisu-Master) [reply src/GlobalSettings.gd]: I adjusted the values to be roughly dpi/96. That's the DPI of my laptop screen and 1.0 looks fine on it, so i used it as a base.

File: src/GlobalSettings.gd (hunk 3 above)
[the remark this answers is not in the document]
Comment (MewPurPur) [reply src/GlobalSettings.gd:260]: Oh, right, that'd change the logic. Then could you just put spaces around the math operators, as per GDScript's style guide?

File: src/GlobalSettings.gd (hunk 2 above)
Comment (aaronfranke) [reply src/GlobalSettings.gd]: Different perspectives I guess, I am using a high DPI desktop display.

File: src/GlobalSettings.gd (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kiisu-Master) [reply src/GlobalSettings.gd:260]: I investigated the function. It does have floor in it but it adds 0.5 which makes it round.
```cpp
double Math::snapped(double p_value, double p_step) {
	if (p_step != 0) {
		p_value = Math::floor(p_value / p_step + 0.5) * p_step;
	}
	return p_value;
}
```

File: src/GlobalSettings.gd (hunk 2 above)
Comment (Kiisu-Master) [reply src/GlobalSettings.gd]: Your DPI will return 2.0 now. 

I found that changing the scaling in Windows also changes the reported DPI. With that, it looks like scaling in GodSVG now roughly matches the OS scaling so it should be good now.

File: src/GlobalSettings.gd [hunk 4]
```diff
@@ -247,26 +246,67 @@ func get_validity_color(error_condition: bool, warning_condition := false) -> Co
...
+	
+	var dpi :=  DisplayServer.screen_get_dpi(screen)
+	if dpi != 72:
+		if dpi < 72:
+			return 0.75
+		elif dpi <= 96:
+			return 1.0
+		elif dpi <= 120:
+			return 1.25
+		elif dpi <=160:
```
Comment (aaronfranke) [inline src/GlobalSettings.gd:296]:
```suggestion
		elif dpi <= 160:
```

**Review by aaronfranke — APPROVED** [review]

**Comment by MewPurPur** [comment]
Thanks A TON!

**Pull request merged**